File: android/app/src/main/java/com/buttery/app/domain/Recipe.kt
package com.buttery.app.domain

data class Recipe(
    val id: Long,
    val title: String,
    val notes: String,
    val prepTime: String,
    val cookTime: String,
    val totalTime: String,
    val servings: String,
    val ingredients: String,
    val instructions: String,
    val photoUri: String?,
    val photoUris: List<String>,
    val videoUri: String?,
    val imageUrl: String?,
    val sourceUrl: String?,
    val originalRawText: String,
    val albumId: Long?,
    val createdAt: Long,
    val updatedAt: Long,
    val isFavorite: Boolean = false,
    val visibility: RecipeVisibility = RecipeVisibility.Private,
    val likeCount: Int = 0,
    val publicPublishedAt: Long? = null
)

data class RecipeAlbum(
    val id: Long,
    val name: String,
    val createdAt: Long,
    val customCoverImageUri: String?
)

enum class RecipeVisibility(val value: String) {
    Private("private"),
    Public("public"),
    Shared("shared");

    companion object {
        fun from(value: String?): RecipeVisibility =
            entries.firstOrNull { it.value.equals(value, ignoreCase = true) } ?: Private
    }
}

File: ios/Shared/Models/Recipe.swift
import Foundation

enum RecipeVisibility: String, Codable, Hashable, CaseIterable {
    case `private`
    case `public`
    case shared
}

struct Recipe: Codable, Hashable, Identifiable {
    let id: String
    let ownerId: String
    var title: String
    var notes: String
    var prepTime: String
    var cookTime: String
    var totalTime: String
    var servings: String
    var ingredients: String
    var instructions: String
    var photoUrls: [URL]
    var videoUrl: URL?
    var sourceUrl: URL?
    var originalRawText: String
    var albumId: String?
    var isFavorite: Bool
    var createdAt: Int64
    var updatedAt: Int64
    var schemaVersion: Int = 1
    var visibility: RecipeVisibility?
    var ownerUsername: String?
    var ownerDisplayName: String?
    var ownerProfilePhotoUrl: String?
    var videoUrls: [URL]?
    var thumbnailUrl: URL?
    var publicPublishedAt: Int64?
    var saveCount: Int?
    var likeCount: Int?
    var viewCount: Int?
    var ratingAverage: Double?
    var ratingCount: Int?
    var originalCreatorId: String?
    var originalCreatorUsername: String?
}

struct RecipeAlbum: Codable, Hashable, Identifiable {
    let id: String
    let ownerId: String
    var name: String
    var customCoverImageUrl: URL?
    var createdAt: Int64
    var updatedAt: Int64
    var schemaVersion: Int = 1
}
